src/foam/core/CompoundContextAgency.js

```diff
@@ -42,7 +43,7 @@ for ( Runnable agent : getAgents() ) {
     e.add(t);
   }
 }
-e.maybeThrow();`
+e.maybeThrow((AppConfig) getX().get("appConfig"));`
```

Comment:
replace this with the context please. 
```suggestion
e.maybeThrow(getX());`
```